File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/AtMostOneOf.java
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Constraint(validatedBy = AtMostOneOfValidator.class)
public @interface AtMostOneOf {
    String message() default "{no.nav.oppgave.AtMostOneOf}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

    String[] fields();

}

File: vtp-pensjon-application/src/main/kotlin/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/AtMostOneOf.kt
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@Suppress("unused")
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Constraint(validatedBy = [AtMostOneOfValidator::class])
annotation class AtMostOneOf(
    val message: String = "{no.nav.oppgave.AtMostOneOf}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
    val fields: Array<String>
)
